pkg/wallet/wallet.go

```diff
@@ -286,16 +288,27 @@ func loadFile(filePath string) (Wallet, error) {
 // Generate instantiates a new wallet, protects its private key and persists it.
 // Everything is dynamically generated except from the nickname.
 func Generate(nickname string, password string) (*Wallet, error) {
-	publicKey, privateKey, err := ed25519.GenerateKey(nil)
+
+	entropy, err := bip39.NewEntropy(256)
+	if err != nil {
+		return nil, fmt.Errorf("creating entropy for mnemonic creation: %w", err)
+	}
+
+	mnemonic, err := bip39.NewMnemonic(entropy)
 	if err != nil {
-		return nil, fmt.Errorf("generating ed25519 keypair: %w", err)
+		return nil, fmt.Errorf("creating mnemonic: %w", err)
 	}
+	fmt.Println("mnemonic")
+	fmt.Println(mnemonic)
+	privateKey := bip39.NewSeed(mnemonic, "")
```

Comment:
it is discouraged to directly use seed as a private key

```suggestion
	seed := bip39.NewSeed(mnemonic, "")
	privateKey := sha512.Sum512(seed)
```